app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-finder-tracker.js

```diff
@@ -0,0 +1,137 @@
+        defaultValue = section.querySelector('input[type=radio]:first-of-type')
+
+        if (eventTarget.id === defaultValue.id) {
+          // Radio elements being reverted to their first option (i.e. their default value) count as a "removed filter".
+          wasFilterRemoved = true
+        }
+      } else if (elementType === 'select') {
+        // The value of a <select> is the value attribute of the selected <option>, which is a hyphenated key. We need to grab the human readable label instead for tracking.
+        elementValue = eventTarget.querySelector("option[value='" + eventTarget.value + "']").textContent
+        defaultValue = eventTarget.querySelector('option:first-of-type').textContent
+
+        if (elementValue === defaultValue) {
+          // <select> elements being reverted to their first option (i.e. their default value) count as a "removed filter". (This will be used on the filter <select>s but not the sort by <select>, as you can't "remove" the sort by filter.)
+          wasFilterRemoved = true
+        }
+      } else if (elementType === 'text') {
+        elementValue = eventTarget.value
+        if (elementValue === '') {
+          // If our custom date filters are reset, they become an empty text box, so we count this as a "removed filter". This boolean won't be used for the keyword search box, as deleting the keyword isn't considered removing a filter.
+          wasFilterRemoved = true
+        }
+      }
+
+      schema.event_data.text = elementValue
+
+      switch (changeType) {
+        case 'update-filter':
+          schema.event_data.event_name = 'select_content'
+
+          if (section) {
+            schema.event_data.section = section.getAttribute('data-ga4-section')
+          }
+
+          if (wasFilterRemoved) {
+            schema.event_data.action = 'remove'
+            schema.event_data.text = elementType === 'text' ? undefined : elementValue
+          } else {
+            schema.event_data.action = elementType === 'text' ? 'search' : 'select'
+            schema.event_data.index = this.getSectionIndex(section)
+          }
+          break
+
+        case 'update-keyword':
+          schema.event_data.event_name = 'search'
+          schema.event_data.url = window.location.pathname
+          schema.event_data.section = 'Search'
+          schema.event_data.action = 'search'
+          schema.event_data.text = PIIRemover.stripPIIWithOverride(schema.event_data.text, true, true)
+          break
+
+        case 'clear-all-filters':
+          schema.event_data.event_name = 'select_content'
+          schema.event_data.action = 'remove'
+          schema.event_data.text = 'Clear all filters'
+          break
+
+        case 'update-sort':
+          schema.event_data.event_name = 'select_content'
+          schema.event_data.action = 'sort'
+          schema.event_data.section = 'Sort by'
+          break
+
+        default:
+          break
```

Comment:
Not sure this needs a default if all it does is `break`?

Reply:
Good spot, I think JS was complaining when I was building the switch statement so I added it while debugging!